readme.md

```diff
@@ -198,39 +198,89 @@ Uses [`feather-icons`](https://github.com/feathericons/feather) under the hood.
 ```
 
 ### ListManager
+ListManager handles
+- filtering and sorting
+- pagination (auto/manual)
+- handle list item updates
+- options based style rendering
+- handle item click and selection events
+
+The component exposes `manager` which has the following methods: `loadMore` and `update` and also to access `selectedItems`
+
+`manager.loadMore()` loads more list items by calling the list api using the provided options
+
+`manager.update()` can be used to change the filters and sort_by options that are applied on the list
+```js
+manager.update({
+  filters: { status: 'Open' },
+  order_by: 'modified desc'
+})
+```
+
+option based rendering?
+class tags for each template can be passed via options, eg:
+```js
+options: {
+  style: {
+    header: 'space-x-2',
+    body: 'space-y-2',
+    row: {
+      base: 'space-x-2 hover:bg-gray-50',
+      selected: 'bg-blue-50 hover:bg-blud-100'
+    },
+    footer: ''
+  }
+}
+```
+
+Event handling
+- selection events should be explicityly overrided via `select` method in `<template #listItem="{ row, select }">`
+- if an item is selected then the on_click for the row is overrided for selection based events
+- a callback can be passed to `handle_row_click` options, which gets triggered when the row is clicked
+
+
 ```html
 <ListManager
   ref="list"
   :options="{
-    cache: ['Ticket', 'Desk'],
     doctype: 'Ticket',
     filters: {'status': 'Open'},
     fields: ['name', 'subject', 'status'],
     limit: 20,
     order_by: 'modified desc',
-    classes: {
-      body: 'space-y-2'
+    style: {
+      header: 'flex flex-row space-x-2',
+      body: 'space-y-2',
+      row: {
+        base: 'flex flex-row space-x-2',
+        selected: ''
+      }
```

Comment:
I think the List component should not render any HTML. It should be renderless.

Reply:
> I think the List component should not render any HTML. It should be renderless.

I am using resource manager with type list within LIstManager component,
the style of the list can be fully customized by the user, by passing class tags to style options

since this component also handles auto pagination, item selection, click events, etc... I don't know how this could be handled without the list rendering

the renderless part could be handled using row resource manager too,